src/Transport/Creation/AzureServiceBusSubscriptionCreatorV6.cs

```diff
@@ -39,15 +39,15 @@ public AzureServiceBusSubscriptionCreatorV6(ReadOnlySettings settings)
 
             var subscriptionDescription = new SubscriptionDescription(topicPath, subscriptionName);
             
-            // check the if the subscription with event name only is the one we should delete, i.e. event with another namespace owns it
-            if (!await SubscriptionIsReusedAcrossDifferentNamespaces(subscriptionDescription, sqlFilter, namespaceManager).ConfigureAwait(false))
+            // Check subscription with event name only is the one we should delete. If it's reused, then we need to use event full name.
+            if (await SubscriptionIsReusedAcrossDifferentNamespaces(subscriptionDescription, sqlFilter, namespaceManager).ConfigureAwait(false))
```

Comment:
If the was the same event from a different namespace, then the subscription entity we're interested in is using a full type name, rather than just a name. This was a bug. We did the inverse.